### ISSUE TITLE
Issue against Django `master`, especially Django Guardian

### ISSUE THREAD

**Issue by carltongibson** [tomchristie/django-rest-framework#3406]
Running against Django `master` there are a **lot** of issues. 

```
1 warnings, 265 error in 137.92 seconds
```

Most of these are caused by Django Guardian. 

Excluding that from the requirements leads to something more reasonable: 

```
5 failed, 239 passed, 1 warnings, 21 error in 11.27 seconds
```

Only two types of failure: 

```
tests/test_utils.py:153: in setUp
    self.get_model = rest_framework.utils.model_meta.models.get_model
E   AttributeError: module 'django.db.models' has no attribute 'get_model'

...

tests/browsable_api/test_browsable_api.py:36: in test_login_shown_when_logged_out
    self.assertContains(response, '>Log in<')
.tox/py35-djangomaster/lib/python3.5/site-packages/django/test/testcases.py:398: in assertContains
    msg_prefix + "Couldn't find %s in response" % text_repr)
E   AssertionError: False is not true : Couldn't find '>Log in<' in response
```

Errors are all these:

```
.tox/py35-djangomaster/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:682:
...
E   TypeError: Call constructor takes either 0 or 3 positional arguments

...
tests/test_filters.py:11: in <module>
    from django.utils import unittest
E   ImportError: cannot import name 'unittest'
```

I don't think there's much work there but it's masked by the Guardian issues.

**Comment by tomchristie** [comment]
3.3 branch resolves much of this: https://github.com/tomchristie/django-rest-framework/pull/3342

**Comment by carltongibson** [comment]
OK. Most of this is the issue with Django Guardian then... #3342 still has that bucket load of failures against `master`. 

It would be nice to not have those — I'll have a look and see if there's a simple way to exclude a single requirement for the `djangomaster` envs

**Comment by carltongibson** [comment]
OK. Can do this sort of thing: 

```
[testenv:py35-djangomaster] # Specify explicitly to exclude Django Guardian against Django master
deps =
       https://github.com/django/django/archive/master.tar.gz
       -rrequirements/requirements-testing.txt
       markdown==2.5.2
       django-filter==0.10.0
```

Will wait on #3405 so as not to repeat myself rebasing everywhere.

**Comment by carltongibson** [comment]
Fixed in v3.3 branch (by #3427)
